SimDataFormats/GeneratorProducts/interface/ScaleWeightGroupInfo.h

```diff
@@ -12,65 +12,79 @@ namespace gen {
     std::vector<size_t> muIndices_;
     bool containsCentral_ = false;
     int lhaid_ = -1;
+    bool hasAllWeights = false;
     // Dyn_scale
     std::vector<std::string> dynNames_;
     std::vector<std::vector<size_t>> dynVec_;
 
-    void setMuRMuFIndex(WeightMetaInfo& info, float muR, float muF);
-    void setMuRMuFIndex(WeightMetaInfo& info, float muR, float muF, size_t dynNum, std::string dynName);
-    inline int getIndex(int muR, int muF) const { return 3 * muR + muF; }
+    inline int indexFromMus(int muR, int muF) const { return 3 * muR + muF; }
     inline bool isValidValue(float mu) const { return mu == 0.5 || mu == 1.0 || mu == 2.0; }
 
+    enum scaleIndices {
+      muR0p5_muF0p5_idx = 0,
+      muR0p5_muF1_idx = 1,
+      muR0p5_muF2_idx = 2,
+      muR1_muF0p5_idx = 3,
+      Central_idx = 4,
+      muR1_muF2_idx = 5,
+      muR2_muF0p5_idx = 6,
+      muR2_muF1_idx = 7,
+      muR2_muF2_idx = 8
+    };
+
   public:
     static const unsigned int MIN_SCALE_VARIATIONS = 9;
     ScaleWeightGroupInfo() : ScaleWeightGroupInfo("") {}
     ScaleWeightGroupInfo(std::string header, std::string name)
-        : WeightGroupInfo(header, name), muIndices_(9, -1), dynVec_(9) {
+        : WeightGroupInfo(header, name), muIndices_(MIN_SCALE_VARIATIONS, -1), dynVec_(MIN_SCALE_VARIATIONS) {
       weightType_ = WeightType::kScaleWeights;
       isFunctionalFormVar_ = false;
+      isWellFormed_ = true;
     }
     ScaleWeightGroupInfo(std::string header) : ScaleWeightGroupInfo(header, header) {}
     ScaleWeightGroupInfo(const ScaleWeightGroupInfo& other) { copy(other); }
     virtual ~ScaleWeightGroupInfo() override {}
     void copy(const ScaleWeightGroupInfo& other);
     virtual ScaleWeightGroupInfo* clone() const override;
     bool containsCentralWeight() const { return containsCentral_; }
+    void addContainedId(int globalIndex, std::string id, std::string label, float muR, float muF);
+    bool isWellFormed() { return isWellFormed_ && hasAllWeights; }
```

Comment:
See above comment. I guess the large question is whether a scale group having all 9 variations should be under the purview of the `isWellFormed` function. Because the `muIndices_` vector is sized to have 9 indices, its not trivial to check if all 9 indices have not -1 values

Reply:
I think isWellFormed should indicate that there are 9 properly parsed variations.